src/modules/modal/modal.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useRef, useCallback } from 'react'
+import FocusLock from 'react-focus-lock'
+import ScrollLock from 'react-scrolllock'
+import PropTypes from 'prop-types'
+
+import styles from './modal.css'
+import ModalTitle from './title'
+import ModalFooter from './footer'
+import ModalContent from './content'
+import ModalPortal from './portal'
+
+import { classNames } from 'utils'
+
+const checkAccessibility = (props, propName, componentName) => {
+  if (!props['aria-label'] || !props['aria-labelledby']) {
+    return new Error(
+      `One of props 'aria-label' or 'aria-labelledby' isRequired in '${componentName}'.`
+    )
+  }
+
+  if (!props['aria-label'] && !props['aria-labelledby']) {
+    return new Error(
+      `One of props 'aria-label' or 'aria-labelledby' was not specified in '${componentName}'.`
+    )
+  }
+
+  if (props['aria-label'] && props['aria-labelledby']) {
+    return new Error(
+      `Props 'aria-label' and 'aria-labelledby' in '${componentName}' mutually exclusive.`
+    )
+  }
+
+  return true
+}
```

Comment:
1. Needs to be handled in the upper projects.  Probably with this plugin https://www.npmjs.com/package/babel-plugin-transform-react-remove-prop-types if it is not used already.

Reply:
If you check the [prop-types index file](https://github.com/facebook/prop-types/blob/master/index.js), you will find that the packages ships a simple object and a function to the production to handle all the possible error cases.

The thing you suggest is an extra optimisation but the package is safe and optimised to use in production itself. This is what I meant.

Personally, I would prefer [forcing this accessibility using the Context API](https://github.com/oacore/design/issues/322), what I proposed recently but it was implicit since I used a slightly different example.